src/antares/model/study.py

```diff
@@ -264,9 +264,18 @@ def create_link(
         area_to: str,
         properties: Optional[LinkProperties] = None,
         ui: Optional[LinkUi] = None,
-        existing_areas: Optional[MappingProxyType[str, Area]] = None,
     ) -> Link:
-        link = self._link_service.create_link(area_from, area_to, properties, ui, existing_areas)
+        missing_areas = [area for area in [area_from, area_to] if area not in self._areas]
+        if missing_areas:
+            raise LinkCreationError(area_from, area_to, f"{', '.join(missing_areas)} does not exist")
+
+        existing_link = next(
```

Comment:
We should sort the areas to perform the check as we cannot have 2 links a -> b and b -> a.

Also, I just realized that I think we handle the MAJ badly. As we don't force lower case inside the property name inside the link object :/ + we use these values as if they were ids even they aren't

I think we should introduce a static method that returns the 2 names transformed in ids to use them internally. With this we could use this method here to check if the link already exists and this time we'll be sure the check works.

Could be done inside a next PR